### DISCUSSION TITLE
Installing edited code with pyproject.toml

### DISCUSSION THREAD

**Discussion by HuangJiameng** [deepmodeling/dpgen#1047] [General]
When I try to install `dpgen` with edited source code using the command `pip install .`, the following ERROR occurs.
```
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 /usr/lib/python3/dist-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-34etvn_j/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i http://mirrors.cloud.aliyuncs.com/pypi/simple/ --trusted-host mirrors.cloud.aliyuncs.com -- oldest-supported-numpy 'setuptools>=43.0.0' 'Cython>=0.29.23'
...
  ERROR: Could not install packages due to an EnvironmentError: [Errno 28] No space left on 
```
When I try another machine, another error occurs.
```
        File "/tmp/pip-standalone-pip-rsgk656_/__env_pip__.zip/pip/_vendor/urllib3/response.py", line 440, in _error_catcher
          raise ReadTimeoutError(self._pool, None, "Read timed out.")
      pip._vendor.urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Read timed out.
```

I am not sure if I am installing it in the right way, so I search for https://stackoverflow.com/questions/62983756/what-is-pyproject-toml-file-for
And use `pip install -e .` to install with `setup.py`, which has the following contents:
```
import setuptools

if __name__ == "__main__":
    setuptools.setup()
```
 However, it doesn't work.
I just want to ensure I have done it right, and if so, I will check my machine then.

**Comment by njzjz** [comment]
Consider upgrading your pip.

**Comment by HuangJiameng** [comment]
After upgrading pip, problem still exist... `pip._vendor.urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Read timed out.`

**Reply by njzjz** [reply]
This looks like a network issue.
